internal/database/offline/sqlutil/export.go

```diff
@@ -26,6 +26,49 @@ func Export(ctx context.Context, db *sqlx.DB, opt offline.ExportOpt, backend typ
 		Backend: backend,
 		SqlxDB:  db,
 	}
+	for groupID, snapshotTable := range opt.SnapshotTables {
+		if snapshotTable == types.TempTable {
+			tableName := dbutil.TempTable(fmt.Sprintf("snapshot_%d", groupID))
+			features := opt.Features[groupID]
+			entity := features[0].Entity()
+
+			schema := ""
+			// export will failed since mysql will trigger an error when you refer to a temporary table more than one time in a single query
+			// see: https://dev.mysql.com/doc/refman/8.0/en/temporary-table-problems.html
+			if backend == types.BackendMySQL {
+				schema = dbutil.BuildTableSchema(tableName, entity, true, features, []string{entity.Name}, backend)
+			} else {
+				schema = dbutil.BuildTempTableSchema(tableName, entity, true, features, []string{entity.Name}, backend)
+			}
+			if err := dbOpt.ExecContext(ctx, schema, nil); err != nil {
+				return nil, err
+			}
+
+			opt.SnapshotTables[groupID] = tableName
+		}
+	}
+
+	for groupID, cdcTable := range opt.CdcTables {
+		if cdcTable == types.TempTable {
+			tableName := dbutil.TempTable(fmt.Sprintf("cdc_table_%d", groupID))
+			features := opt.Features[groupID]
+			entity := features[0].Entity()
+
+			schema := ""
+			// export will failed since mysql will trigger an error when you refer to a temporary table more than one time in a single query
+			// see: https://dev.mysql.com/doc/refman/8.0/en/temporary-table-problems.html
+			if backend == types.BackendMySQL {
+				schema = dbutil.BuildTableSchema(tableName, entity, true, features, []string{entity.Name}, backend)
+			} else {
+				schema = dbutil.BuildTempTableSchema(tableName, entity, true, features, []string{entity.Name}, backend)
+			}
```

Comment:
maybe wrap the `if else` statement in the `BuildTempTableSchema` function?

Reply:
execute could also be wrapped into `BuildTempTableSchema`

Reply:
I was thinking of putting it in BuiltTempTableSchema to get away from the first site and also to make BuiltTempTableSchema not necessarily a build temporary table anymore.